AirCasting/SessionViews/SessionCartViewModel/DefaultDeleteSessionViewModel.swift

```diff
@@ -4,14 +4,87 @@
 import Foundation
 
 class DefaultDeleteSessionViewModel: DeleteSessionViewModel {
+    let measurementStreamStorage: MeasurementStreamStorage
+    var session: SessionEntity
+    let streamRemover: StreamRemover
+    
     var deleteEnabled: Bool = false {
         willSet {
             objectWillChange.send()
         }
     }
     
+    // It is variable because of the following scenario :
+    // 1. Go to delete session scene
+    // 2. Remove one of the streams on different device
+    // 3. It should after sync it should get removed from the view
+    
+    var options: [DeleteSessionOptionViewModel] {
+        willSet {
+            objectWillChange.send()
+        }
+    }
+    
+    init(session: SessionEntity, measurementStreamStorage: MeasurementStreamStorage, streamRemover: StreamRemover) {
+        
+        self.measurementStreamStorage = measurementStreamStorage
+        self.session = session
+        self.streamRemover = streamRemover
+        
+        var sessionStreams: [MeasurementStreamEntity] {
+            return session.sortedStreams?.filter( {!$0.gotDeleted} ) ?? []
+        }
+        
+        options = [.init(id: -1, title: "All", isSelected: false, isEnabled: false)]
+        showProperStreams(sessionStreams: sessionStreams)
+    }
+    
+    func showProperStreams(sessionStreams: [MeasurementStreamEntity]) {
+        for (id, stream) in sessionStreams.enumerated() {
+            if var streamName = stream.sensorName {
+                if streamName == Constants.SensorName.microphone {
+                    options.append(.init(id: id, title: "dB", isSelected: false, isEnabled: false))
+                } else {
+                    streamName = streamName.components(separatedBy: "-")[1]
```

Comment:
I would guard against the count of those components. I know it *should* always be in format `asd-asd`, but this is a string which we get directly from an external source (webservice) so we should be extra careful.

Reply:
The same concept is used trough the whole app as I know. Should I change it somehow, place a warning? What would you suggest?

Reply:
I it's used in more than one place I'd create an extension to deal with it (and add this guard there)